modules/appeals_api/lib/appeals_api/higher_level_review_pdf_constructor.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class HigherLevelReviewPdfConstructor
+    PDF_TEMPLATE = Rails.root.join('modules', 'appeals_api', 'config', 'pdfs')
+
+    def initialize(higher_level_review_id)
+      @higher_level_review = AppealsApi::HigherLevelReview.find(higher_level_review_id)
+      veteran
+    end
+
+    def veteran
+      @veteran ||= build_veteran(@higher_level_review)
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def build_veteran(higher_level_review)
+      form_data = higher_level_review.form_data['data']['attributes']
+      included = higher_level_review.form_data['included']
+      auth_headers = higher_level_review.auth_headers
+      OpenStruct.new(
+        first_name: auth_headers['X-VA-First-Name'],
+        middle_name: auth_headers['X-VA-Middle-Initial'],
+        last_name: auth_headers['X-VA-Last-Name'],
+        ssn: auth_headers['X-VA-SSN'],
+        va_file_number: auth_headers['X-VA-File-Number'],
+        service_number: auth_headers['X-VA-Service-Number'],
+        insurance_policy_number: auth_headers['X-VA-Insurance-Policy-Number'],
+        birth_date: auth_headers['X-VA-Birth-Date'],
+        address_line_1: form_data.dig('veteran', 'address', 'addressLine1'),
+        address_line_2: form_data.dig('veteran', 'address', 'addressLine2'),
+        city: form_data.dig('veteran', 'address', 'city'),
+        state: form_data.dig('veteran', 'address', 'stateCode'),
+        country: form_data.dig('veteran', 'address', 'country'),
+        zip: form_data.dig('veteran', 'address', 'zipCode'),
+        zip_last_4: form_data.dig('veteran', 'address', 'zip_last_4'),
+        benefit_type: form_data['benefitType'],
+        same_office: form_data['sameOffice'],
+        informal_conference: form_data['informalConference'],
+        conference_times: form_data['informalConferenceTimes'],
+        issues: included
+      )
+    end
```

Comment:
Do you think it worthwhile to make a Veteran model that can contain this business logic?

Reply:
we can refactor later but we already have a few veteran models that have this sort of information that we can bringin and update etc.